### DISCUSSION TITLE
ViewStore.init on iOS, objectWillChange error?

### DISCUSSION THREAD

**Discussion by sbeitzel** [pointfreeco/swift-composable-architecture#1457] [Beta]
I've got a macOS app well-integrated with TCA, and even up to date on the ReducerProtocol and Dependency modifications, so I thought I'd try making an iOS version of the app. After creating the target and firing it up just to see how bad the UI looked, I got an unexpected warning.

"Publishing changes from within view updates is not allowed, this will cause undefined behavior." This is coming from ViewStore.init on line 90, where the initializer calls `objectWillChange()`, and it seems to be triggered by tapping the login button:

```swift
            Button(action: { viewStore.send(.login) },
                   label: { Text(LocalizedStringKey.avLogin) })
            .disabled(viewStore.authState.email.isEmpty || viewStore.authState.password.isEmpty)
```

Further poking around reveals that this warning happens any time the user pokes anything in the UI that triggers an action being sent. But, notably, this is only on iOS, even though at this point I haven't got so far as to create platform-specific Views. Running the exact same code built for macOS doesn't generate this warning. Interestingly, it also doesn't happen when I run the iOS app (built for iPad) on my mac. On iOS, I've only tried this on a simulator, not on a real device, so I'm not sure if this is an iOS bug or a simulator bug, or if this is something to do with the way TCA gets compiled for iOS.

Has anyone else seen this?

**Comment by jshier** [comment]
This is a bug in Xcode 14. It should be fixed if you build and run using Xcode 14.1b3. There should be other reports of the same issue in this repo.

**Comment by tgrapperon** [comment]
@sbeitzel Indeed, this is a bug. You can find more information at #1319. As @jshier mentioned, this will be fixed with the next version of Xcode.
